### PR TITLE
Feature/create agnostic services

### PR DESCRIPTION
**Pull request by pgarciaegido**
The main idea of this PR is to start taking out logics from React to agnostic services that could be used regarding of the framework in use.

In this PR I have created three services:
- `crossmintModalService`. This replaces the existing hook `useCrossmintModal`.
- `crossmintPayButtonService`. This would take care of all the logics that happen on the `CrossmintPayButton` view.
- `crossmintStatusService`. It would take care of the logics regarding fetching the status and handling the button status view.

This was actually pretty straight forward since our components are not really too heavy on reactivity, so it is fairly easy to take the logics out.

I would like to get an initial feedback on this approach, and if I get the green light Ill move all the logics into common services and start creating Vue, Angular and Svelte wrappers this weekend.

Screenshot 2022-03-04 at 18 38 24

Thank you!

### CODE REVIEW

File: packages/ui/react-ui/__tests__/CrossmintPayButton.spec.tsx [hunk 1]
```diff
@@ -14,31 +14,83 @@ global.fetch = jest.fn(() => fetchReturns);
 // TODO(#61): make this automatically mocked in every test suite
```
Comment (pgarciaegido) [inline packages/ui/react-ui/__tests__/CrossmintPayButton.spec.tsx:14]: IGNORE THIS FILE.

This has its own PR. https://github.com/Crossmint/crossmint-client-sdk/pull/63

**Comment by pgarciaegido** [comment]
Tests are failing because now its depending on the base package... Will see how to fix later

**Comment by mPaella** [comment]
I don't think that the hooks should go inside the base package? 

If u checkout [the example this repo is based on](https://github.com/solana-labs/wallet-adapter/tree/master/packages/core), they have separate folders for react, and base is a general package containing only non-framework specific types and functions.

**Comment by pgarciaegido** [comment]
There are no hooks inside the base package. 

Maybe I should change the way the services are used. Instead of:

`const { connect } = useCrossmintModalService();`

Im changing it for:

`crossmintModalService.connect()`;

So its clear that there is no reactivity in the base logics, just pure typescript.

File: packages/ui/react-ui/src/CrossmintPayButton.tsx [hunk 2]
```diff
@@ -30,34 +33,34 @@ export const CrossmintPayButton: FC<CrossmintPayButtonProps> = ({
...
-            "No collection description specified. Please add a collection description to your <CrossmintPayButton />"
-        );
-        collectionDescription = "";
-    }
-
-    if (collectionPhoto === "<OPT_URL_TO_PHOTO_COVER>") {
-        console.warn("No collection photo specified. Please add a collection photo to your <CrossmintPayButton />");
-        collectionPhoto = "";
-    }
+    const [newCollectionTitle, newCollectionDescription, newCollectionPhoto] = checkProps({
```
Comment (mPaella) [inline packages/ui/react-ui/src/CrossmintPayButton.tsx:49]:
```
[collectionTitle, collectionDescription, collectionPhoto] = checkProps({
        collectionTitle,
        collectionPhoto,
        collectionDescription,
    });
```

works?

**Comment by mPaella** [comment]
> There are no hooks inside the base package.
> 
> Maybe I should change the way the services are used. Instead of:
> 
> `const { connect } = useCrossmintModalService();`
> 
> Im changing it for:
> 
> `const { connect } = crossmintModalService();`
> 
> So its clear that there is no reactivity in the base logics, just pure typescript. Reactivity should be handled at the component level.
> 
> Im [using functions to encapsulate information](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) instead of classes (as in the example you provided). Do classes make more sense to you?
> 
> @mPaella

Functions are good. Only other thing I can see is possibly the useCrossmintStatus hook should go inside the base package as well? wdyt?

**Comment by pgarciaegido** [comment]
cc @angel-paella 

@mPaella could you please test locally to ensure that everything is working correctly.

For the tests to work, we would need to publish the new base package. Locally, they work.

Thanks!

**Comment by mPaella** [comment]
LGTM. We should also add the tsup build to the base package, then deploy both.

**Pull request merged**